crates/btcio/src/reader/handler.rs

```diff
@@ -0,0 +1,163 @@
+use bitcoin::{consensus::serialize, hashes::Hash, Block};
+use secp256k1::XOnlyPublicKey;
+use strata_l1tx::messages::{BlockData, L1Event};
+use strata_primitives::{
+    buf::Buf32,
+    l1::{L1BlockManifest, L1BlockRecord},
+};
+use strata_state::{
+    batch::{BatchCheckpoint, BatchCheckpointWithCommitment, CommitmentInfo},
+    l1::{generate_l1_tx, L1Tx},
+    sync_event::{EventSubmitter, SyncEvent},
+    tx::ProtocolOperation,
+};
+use tracing::*;
+
+use super::query::ReaderContext;
+use crate::rpc::traits::ReaderRpc;
+
+pub(crate) async fn handle_bitcoin_event<R: ReaderRpc, E: EventSubmitter>(
+    event: L1Event,
+    ctx: &ReaderContext<R>,
+    event_submitter: &E,
```

Comment:
Can do `&impl EventSubmitter`, but it would still be better to use an observer-like system.  This is sortaaa decoupled because there's the trait but this is actually just obscuring a tighter dependency.

Reply:
Oh yes, I could have done `&impl EventSubmitter`.

> it would still be better to use an observer-like system

It would possibly be better, but that should be done in a separate ticket, because that requires broader design discussions.

Reply:
Do you think we can have a better name though?

Reply:
The name is fine.

Reply:
Alright I guess since there's more CSM stuff due to be refactored in the mid future we can just leave it as-is.